### DISCUSSION TITLE
List of all known Ring API endpoints

### DISCUSSION THREAD

**Discussion by singerscreations** [dgreif/ring#759] [General]
Is there a list of all known Ring API endpoints somewhere? For example, all Ring devices can be retrieved via this endpoint:

https://api.ring.com/clients_api/ring_devices

Is there an endpoint to get a list of devices (and statuses) attached to an alarm base station (base_station_v1)? And how about endpoints to see the status and control Z-Wave light switches attached to an alarm base station?

**Comment by dgreif** [comment]
The alarm and smart lighting do all communication via websockets.  Those interactions can be found in https://github.com/dgreif/ring/blob/master/api/location.ts.  Regular api endpoints are available for cameras and chimes, but they aren't all in one central location in the code here.
